Docs/build_linux.md

```diff
@@ -38,65 +38,41 @@ CARLA requires many different kinds of software to run. Some are built during th
 
 ```sh
 sudo apt-get update &&
-sudo apt-get install wget software-properties-common &&
-sudo add-apt-repository ppa:ubuntu-toolchain-r/test &&
-wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key|sudo apt-key add
+sudo apt-get install wget software-properties-common
 ```
 
 !!! Warning
     The following commands depend on your Ubuntu version. Make sure to choose accordingly. 
 
-To avoid compatibility issues between Unreal Engine and the CARLA dependencies, use the same compiler version and C++ runtime library to compile everything. The CARLA team uses clang-8 (or clang-10 in Ubuntu 20.04) and LLVM's libc++. Change the default clang version to compile Unreal Engine and the CARLA dependencies.
 
 __Ubuntu 22.04__.
 ```sh
-sudo apt-add-repository "deb http://archive.ubuntu.com/ubuntu focal main universe"
-sudo apt-get update
-sudo apt-get install build-essential clang-10 lld-10 g++-7 cmake ninja-build libvulkan1 python python3 python3-dev python3-pip libpng-dev libtiff5-dev libjpeg-dev tzdata sed curl unzip autoconf libtool rsync libxml2-dev git git-lfs
-sudo update-alternatives --install /usr/bin/clang++ clang++ /usr/lib/llvm-10/bin/clang++ 180 &&
-sudo update-alternatives --install /usr/bin/clang clang /usr/lib/llvm-10/bin/clang 180 &&
-sudo update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-7 180
+sudo apt-get install build-essential cmake ninja-build libvulkan1 python-is-python3 python3 python3-dev python3-pip libpng-dev libtiff5-dev libjpeg-dev tzdata sed curl unzip autoconf libtool rsync libxml2-dev git git-lfs
```

Comment:
g++ is required to build modules like ROS2 and pytorch.
Why do you think we should remove it from package installation?

Reply:
only the additional g++-7 is removed (as not required) and the corresponding update alternatives call
build-essentials should bring already the default system g++ with it.

Reply:
Good catch about the build-essential package!

I have addicional worries about this change:
If we define the g++ version as system default one, this implies CARLA could be possibly build with huge number of g++ versions, as many users are building CARLA and we don't have any info about what distro are the using. On my understanding, define g++ version as system default adds the next risks:
 - Sometimes code built with different g++ versions my conclude on build errors or new defects.
 - Is not clear for CARLA contributors what is the minimum requiered version of g++ to mantain build backward compatibility when creating PRs.
 - I'm not sure if ROS2 and pytorch libraries have some restriction about what versions of the comipler to use.
 - It's pracmatic (usable) for the contributors to have a default version of g++ that is tested by us and we ensure is working. Otherwise, they could potentially have to deal with non desired build errors and this will take extra time for them. Anyway, I think we should provide a mechanism to the contributors to build with different versions of g++, but we also should provide a default stable g++ version for the contributors who don't expect to lose time dealing by unestable CARLA build sysmtem.
 
 What are your thoughts about it?

Reply:
I see, yes your fears make sense.
I would then approach from that side. Standard supported linux system is Ubuntu. Then I would go with the active LTS versions and support their standard g++ versions. Taking their EOL dates as supporting dates might be a bit too much as Ubuntu 14.04 goes EOL right now. So taking their "End of Standard Support" as basis might be more meaningful:
Ubuntu 22.04 LTS: June 2027, [gcc-11](https://packages.ubuntu.com/jammy/gcc)
Ubuntu 20.04 LTS: April 2025, [gcc-9](https://packages.ubuntu.com/focal/gcc)
Therefore, I'd either define gcc-9 or gcc-11 as gold standard for now.

On pytorch, I don't know since I have to admit that I never worked so far with it on my own. On ROS2 the respective Ubuntu default compiler for sure works.

But if CARLA supports both active Ubuntu systems (and maybe have a build job for both of them in the background to ensure), then you don't need to install a specific version here. One should then rewrite the docu in the sense that the two active Ubuntu systems are supported. And if someone else with nother Linux-Distro wants to compile CALRA he/she should install either gcc-9 or gcc-11 (at the current point in time for sure ;-)
Still, then you don't need to change the build instructions for 22.04 and 20.04, and why not making the live harder for the restistent guys who are still using 18.04 or older; If they do use such old systems they should be aware of that and be able to handle to install other gcc versions on their own if they run into problems. Or they should just upgrade their system -- which is from system security perspective in each case adviceable.
The supported clang version is in each case clear, since it's the Unreal one... which will also change soon when swithing to Unreal 5.xx engine.
So you see: you might want to extend the notes in the docu around that topic a bit and "degrade" also 18.04 to a not adviced Linux -Distro, since it has some years "on the back" (don't know if one says that also in UK like that ... it's some German slang ;-)

Please discuss with the team and adapt the commit as you decide on your own. You should have the rights to do so, I believe.

Ah, so now it's Easter in Germany: Happy searching of Easter Eggs.

Reply:
Happy Easter!